src/transactions/TransactionFrame.cpp

```diff
@@ -829,12 +829,22 @@ TransactionFrame::applyOperations(SignatureChecker& signatureChecker,
         CLOG_ERROR(Tx, "Exception while applying operations ({}, {}): {}",
                    xdr_to_string(getFullHash(), "fullHash"),
                    xdr_to_string(getContentsHash(), "contentsHash"), e.what());
+        if (app.getConfig().HALT_ON_INTERNAL_TRANSACTION_ERROR)
```

Comment:
no need to duplicate this code: you can just do it later in that function

Reply:
Done.